### PR TITLE
Fixes #29470 - Rails 6 compatible method calls

### PR DESCRIPTION
**Pull request by ezr-ondrej**
Moves to rails 6 syntax and adds a patch file to keep compatibility to Rails 5.2

### CODE REVIEW

File: app/models/concerns/nested_ancestry_common.rb [hunk 1]
```diff
@@ -7,8 +7,8 @@ module NestedAncestryCommon
     has_ancestry :orphan_strategy => :restrict
 
     before_validation :set_title
-    after_save :set_other_titles, :on => [:update, :destroy]
-    after_save :update_matchers, :on => :update, :if => proc { |obj| obj.saved_change_to_title? }
+    after_save :set_other_titles
+    after_save :update_matchers, :if => proc { |obj| obj.saved_change_to_title? }
```
Comment (tbrisker) [inline app/models/concerns/nested_ancestry_common.rb]: This looks like changed logic, should these be `after_update`?
Also, looks like @mmoll already did this change (plus another place) in https://github.com/theforeman/foreman/pull/7322 - let's spin it off to a separate pr cleaning up all the `:on`s and merge that.

Comment (tbrisker) [reply app/models/concerns/nested_ancestry_common.rb]: Extracted this to #7566

Comment (ezr-ondrej) [reply app/models/concerns/nested_ancestry_common.rb]: Awesome, thanks :heart:

File: app/registries/foreman/plugin.rb [hunk 2]
```diff
@@ -345,8 +345,8 @@ def add_all_permissions_to_default_roles
     def pending_migrations
       return true if Foreman.in_setup_db_rake?
       migration_paths = ActiveRecord::MigrationContext.new(
-        ActiveRecord::Migrator.migrations_paths).migrations
-      pending_migrations = ActiveRecord::Migrator.new(:up, migration_paths).
+        ActiveRecord::Migrator.migrations_paths, ActiveRecord::SchemaMigration).migrations
+      pending_migrations = ActiveRecord::Migrator.new(:up, migration_paths, ActiveRecord::SchemaMigration).
         pending_migrations
```
Comment (tbrisker) [inline app/registries/foreman/plugin.rb]: Do we really need all this code? `ActiveRecord::Base.connection.migration_context.needs_migration?` will give the same return value afaict, we would lose the debug logging but i highly doubt it's actually useful at all

Comment (tbrisker) [reply app/registries/foreman/plugin.rb]: here you go:
https://github.com/theforeman/foreman/pull/7571

**Review by tbrisker — APPROVED** [review]
Thanks @ezr-ondrej !
